keras/preprocessing/sequence.py

```diff
@@ -12,6 +12,45 @@
 from ..utils.data_utils import Sequence
 
 
+def make_step_based_generator(x=None, y=None, sample_weights=None,
+                              steps=None):
+    """Provide Step-based generator used by: fit/test/predict
+       when steps/steps_per_epoch is not None.
+
+       1) fit(.., steps_per_epoch=k, ..)  =>
+              gen, steps = make_step_based_generator(..)
+              fit_generator(gen, steps_per_epoch=steps, ..)
+
+       2) evaluate(.., steps=k, ..)  =>
+              gen, steps = make_step_based_generator(..)
+              steps_generator(gen, steps=steps, ..)
+
+       3) predict(.., steps=k, ..)  =>
+              gen, steps = make_step_based_generator(..)
+              predict_generator(gen, steps=steps, ..)
+
+    # Arguments
+        x: Transfered from arguement x in fit/test/predict.
+        y: Transfered from arguement y in fit/test.
+        sample_weights: sample_weights provided from fit/test.
+        steps: steps or steps_per_epoch provided from fit/test/predict.
+
+    # Returns
+        generator: the generator to proceduce step-based data.
+            the output of the generator of one of the following formats:
+            - a tuple `(x, y)`
+            - a tuple `(x, y, sample_weights)`.
+        steps: the number of times the generator is allowed to proceduce data.
+    """
+    def step_based_generator(x, y, sample_weights, steps):
+        for i in range(steps):
+            if sample_weights is None:
+                yield (x, y)
+            else:
+                yield (x, y, sample_weights)
+    return step_based_generator(x, y, sample_weights, steps), steps
+
+
```

Comment:
There can also be `make_batch_based_generator` in the future which will eventually fix #9667, so here, I make function `make_step_based_generator` a little formal in its returning value.